db/db/src/main/java/io/helidon/db/DbExecute.java

```diff
@@ -0,0 +1,407 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.db;
+
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Database executor.
+ * <p>The database executor provides methods to create {@link DbStatement} instances for different types
+ * of database statements.</p>
+ * <p>The recommended approach is to use named statements, as that allows better metrics, tracing, logging etc.
+ * In case an unnamed statement is used, a name must be generated
+ * <p>There are five methods for each {@link DbStatementType}, example for query (the implementation
+ * detail is for the default implementation, providers may differ):
+ * <ol>
+ *     <li>{@code DbStatement} {@link #createNamedQuery(String, String)} - full control over the name and content of the
+ *     statement</li>
+ *     <li>{@code DbStatement} {@link #createNamedQuery(String)} - use statement text from configuration</li>
+ *     <li>{@code DbStatement} {@link #createQuery(String)} - use the provided statement, name is generated </li>
+ *     <li>{@code DbRowResult} {@link #namedQuery(String, Object...)} - shortcut method to a named query with a list of
+ *     parameters (or with no parameters at all)</li>
+ *     <li>{@code DbRowResult} {@link #query(String, Object...)} - shortcut method to unnamed query with a list of parameters
+ *     (or with no parameters at all)</li>
+ * </ol>
+ * The first three methods return a statement that can have parameters configured (and other details modified).
+ * The last two methods directly execute the statement and provide appropriate response for future processing.
+ * All the methods are non-blocking.
+ */
+public interface DbExecute {
+    /*
+     * QUERY
+     */
+
+    /**
+     * Create a database query using a named statement passed as argument.
+     *
+     * @param statementName the name of the statement
+     * @param statement the query statement
+     * @return database statement that can process query returning multiple rows
+     */
+    DbStatement<?, DbRowResult<DbRow>> createNamedQuery(String statementName, String statement);
```

Comment:
Why do we have this kind of hybrid method here? It makes sense to have 
- createNamedQuery with statement name as an argument -> statement is retrieved from statements collection by name
- createQuery with statement string as an argument -> statement is passed directly

We can define an annotation to allow an additional way of adding named statements.  Doing it this way is wrong and does not match behavior of other frameworks (JPA).

Reply:
No annotations in Helidon SE.
The reason is that metrics, tracing and possibly other tools require a name to attach to a statement.